### PR TITLE
chore: Add SDK readme.

### PR DESCRIPTION
**Pull request by fmarek-kindred**

### CODE REVIEW

File: cohort_sdk_client/README.md [hunk 1]
```diff
@@ -1 +1,304 @@
...
+```
+The purpose of statemap installer callback is to "catch-up" on changes made by other cohorts (including your cohort).
+
+In the deployment where there is no Talos present, microservices typically implement eventual consistency model by utilising messaging broker. For example, Kafka. If changes are made to bank account in Cohort 1 that change will eventually propagate to Cohort 2 through messaging middleware.
+
+In the deployment with Talos, the replicator, specifically this callback, is responsible for updating the database so that your cohort has the up-to-date view of bank account (or any other shared object). The async messaging is abstracted away from you by SDK. You just provide a handler how you want your database to be updated.
+
+However, there are few rules which Talos protocols expects you to implement.
+
+- Once you are done updating business objects in your database, it is very important to update the global stapshot. The replicator, specifically this callback, is the only place within cohort which is responsible for writing into snapshot table.
```
Comment (gk-kindred) [inline cohort_sdk_client/README.md:293]: Small typo - 
Update 
`global stapshot` to `global snapshot`

File: cohort_sdk_client/README.md [hunk 2]
```diff
@@ -1 +1,304 @@
...
+- The change to business objects and to snapshot table should be atomic (in a single DB transaction with isolation level matching repeatable read or stricter).
+- When callback is invoked, it is possible that your business objects are already updated. In this case, the job of callback is to update the snapshot table only.
+  - This may happen if replicator and initiator belong to the same cohort, for example, out of order installer in initiator may have executed updated our business objects before the replicator. However, installer should never write to snapshot table.
+  - When replicator belong to different cohort, it is just catching up on the changes made by other cohorts, hence it may not encounter the state when business objects are updated already. Unless there was some contingency, like unexpected restart.
+- When updating business objects, also update their versions so that versions match with snapshot version.
+
+What value to write into snapshot table? Use this attribute: `JsStatemapAndSnapshot.version`
+What version to set into business objects? Use this attribute: `JsStatemapAndSnapshot.version`
+
+## About Statemap
```
Comment (gk-kindred) [inline cohort_sdk_client/README.md:304]: Should I wait for this section to be completed before approving this PR?

File: cohort_sdk_client/README.md (hunk 1 above)
Comment (fmarek-kindred) [reply cohort_sdk_client/README.md:293]: Lol thanks.

File: cohort_sdk_client/README.md (hunk 2 above)
Comment (fmarek-kindred) [reply cohort_sdk_client/README.md:304]: Please approve at any time, I needed your review for the content.

**Review by gk-kindred — APPROVED** [review]
Thanks for taking time to document this. Very good effort!

**Pull request merged**